apps/api/src/app/plugins/bffCache.ts

```diff
@@ -0,0 +1,78 @@
+import fp from "fastify-plugin";
+import { CACHE_CONTROL_HEADER, getCache, getCacheControlHeaderValue, parseCacheControlHeaderValue, setCache } from "../../utils/cache";
+import { FastifyPluginCallback, FastifyReply, FastifyRequest } from "fastify";
+
+const HEADER_NAME = 'x-bff-cache'
+
+interface BffCacheOptions {
+  ttl?: number
+}
+
+export const bffCache: FastifyPluginCallback<BffCacheOptions> = (fastify, opts, next) => {
+  const { ttl } = opts
+  fastify.addHook('onRequest', async (request, reply) => {
+    // Cache only GET requests
+    if (request.method !== 'GET') {
+      return
+    }
```

Comment:
What if the GET request returns an error? Anything that's non 2xx

Reply:
Good point, I will not cache those